### PR TITLE
server: split aparts controller + CRD code

### PR DESCRIPTION
**Pull request by sttts**
/assign @ncdc

### CODE REVIEW

**Comment by ncdc** [comment]
This is going to conflict with #390 - which do you want to go in first?

**Comment by ncdc** [comment]
Maybe #390 makes more sense to go in first, so this PR is smaller?

File: pkg/server/config.go [hunk 1]
```diff
@@ -30,48 +30,48 @@ import (
...
-	Authentication             *kubeoptions.BuiltInAuthenticationOptions
+	EtcdClientInfo            etcd.ClientInfo
+	EtcdDirectory             string
+	EtcdPeerPort              string
+	EtcdClientPort            string
+	EtcdWalSizeBytes          int64
+	InstallClusterController  bool
+	ClusterControllerOptions  *cluster.Options
+	installWorkspaceScheduler bool
+	InstallNamespaceScheduler bool
```
[the remark this answers is not in the document]
Comment (sttts) [reply pkg/server/config.go:61]: fixing

**Comment by s-urbaniak** [comment]
👍 on uncluttering `server.go`

**Review by s-urbaniak — APPROVED** [review]

**Pull request merged**